WooCommerce/Classes/ViewRelated/Dashboard/GoogleAds/GoogleAdsDashboardCardViewModel.swift

```diff
@@ -29,6 +46,73 @@ final class GoogleAdsDashboardCardViewModel: ObservableObject {
     }
 
     func dismissCard() {
-        // TODO
+        analytics.track(event: .DynamicDashboard.hideCardTapped(type: .googleAds))
+        onDismiss?()
+    }
+
+    @MainActor
+    func fetchLastCampaign() async {
+        syncingError = nil
+        syncingData = true
+        analytics.track(event: .DynamicDashboard.cardLoadingStarted(type: .googleAds))
+        do {
+            let campaigns = try await fetchAdsCampaigns()
+            lastCampaign = {
+                // prioritize showing last enabled campaign.
+                let enabledCampaigns = campaigns.filter { $0.status == .enabled }
+                return enabledCampaigns.last ?? campaigns.last
+            }()
+            if let id = lastCampaign?.id {
+                Task {
+                    await updateCampaignStats(campaignID: id)
+                }
+            }
+            analytics.track(event: .DynamicDashboard.cardLoadingCompleted(type: .googleAds))
+        } catch {
+            syncingError = error
+            analytics.track(event: .DynamicDashboard.cardLoadingFailed(type: .googleAds, error: error))
+            DDLogError("⛔️ Error loading Google ads campaigns: \(error)")
+        }
+        syncingData = false
+    }
+}
+
+private extension GoogleAdsDashboardCardViewModel {
+    @MainActor
+    func fetchAdsCampaigns() async throws -> [GoogleAdsCampaign] {
+        try await withCheckedThrowingContinuation { continuation in
+            stores.dispatch(GoogleAdsAction.fetchAdsCampaigns(siteID: siteID) { result in
+                continuation.resume(with: result)
+            })
+        }
+    }
+
+    @MainActor
+    func updateCampaignStats(campaignID: Int64) async {
```

Comment:
Opinion: To treat this as a helper method, we could return the campaign stats total value from this method and assign it to `lastCampaignStats` from the calling place. I think this will improve readability.

Reply:
Good idea! Updated in 7acd489eaf9a4a40b000d589036947122e7a09c8.